passes/cmds/rename.cc

```diff
@@ -92,13 +92,23 @@ static IdString derive_name_from_cell_output_wire(const RTLIL::Cell *cell, strin
 
 		name += chunk.wire->name.str();
 		if (chunk.wire->width != chunk.width) {
-			name += "[";
-			if (chunk.width != 1)
-				name += std::to_string(chunk.offset + chunk.width) + ":";
-			name += std::to_string(chunk.offset) + "]";
+			int lhs = chunk.wire->to_hdl_index(chunk.offset + chunk.width - 1);
+			int rhs = chunk.wire->to_hdl_index(chunk.offset);
+			if (chunk.wire->upto)
```

Comment:
On some level, I would expect printing to respect the uptoness of the original hdl objects, so I don't think we should be undoing it here with the swap

In case it's useful / becomes part of a test, here's the input for trying out this code path
```
read_verilog <<EOF
module top(input [0:7] a, input [0:7] b, output [0:4] o1, output [4:0] o2);
    wire [0:7] int1 = a + b;
    wire [0:7] int2 = a + b;
    assign o1[0:3] = int1[0:3];
    assign o2[3:0] = int2[4:7];
endmodule
EOF

proc -noopt
rename -hide w:int*
opt
rename -wire -move-to-cell
```